### ISSUE TITLE
`inlinevar-rgx` seems to react to scope

### ISSUE THREAD

**Issue by Teagum** [PyCQA/pylint#7969]
### Bug description

pylint seems to ignore `inlinevar-rgx` if iteration variables are defined within functions. Given the below configuration, pylint throws false-positive C0103 errors for each iteration variable.

```
# v.py
# pylint: disable=missing-docstring

def func():
    index = ((1, 2), (2, 2))
    for (wx, wy), (ix, iy) in zip(index, index):
        print(wx, wy, ix, iy)
```
However, if you define the same loop on the module top level, there is no error, as expected.

```
# v1.py
# pylint: disable=missing-docstring

index = ((1, 2), (2, 2))
for (wx, wy), (ix, iy) in zip(index, index):
    print(wx, wy, ix, iy)
```

### Configuration

```ini
[BASIC]
inlinevar-rgx=[a-z][a-z]+$
```

### Command used

```shell
$ pylint v.py
```

### Pylint output

```shell
************* Module v
v.py:6:9: C0103: Variable name "wx" doesn't conform to snake_case naming style (invalid-name)
v.py:6:13: C0103: Variable name "wy" doesn't conform to snake_case naming style (invalid-name)
v.py:6:19: C0103: Variable name "ix" doesn't conform to snake_case naming style (invalid-name)
v.py:6:23: C0103: Variable name "iy" doesn't conform to snake_case naming style (invalid-name)
```

### Expected behavior

No error, regardless of defining scope

### Pylint version

```shell
pylint 2.15.5
astroid 2.12.12
Python 3.10.1 (main, Jan  6 2022, 14:35:41) [Clang 13.0.0 (clang-1300.0.29.30)]
```

### OS / Environment

macOS 11.6.8

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue. A for loop is not inlined so ``inlinevar-rgx`` isn't applied in any of the provided example. invalid name might not be working correctly for module level variable, I'm going to close as a duplicate of #7305
